src/exec.cpp

```diff
@@ -828,10 +829,14 @@ static launch_result_t exec_process_in_job(parser_t &parser, process_t *p,
 
     // Maybe trace this process.
     // TODO: 'and' and 'or' will not show.
-    if (trace_enabled(parser)) {
-        trace_argv(parser, nullptr, p->argv());
+
+    ::rust::Vec<wcharz_t> rust_argv;
```

Comment:
This pattern is duplicated three times; can we add a C++ wrapper here? Also the wrapper could check if tracing is enabled before allocating this new vactor.

Reply:
I've now introduced a `to_rust_string_vec` function and a `rust_trace_if_enabled` wrapper.